.gitbook/assets/ReadonlyREST-api.yml

```diff
@@ -0,0 +1,209 @@
+      responses:
+        200:
+          description: Returns a status of updating Main Settings
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UpdateMainSettingsResponse'
+              examples:
+                UpdateMainSettingsSuccessResponse:
+                  $ref: "#/components/examples/UpdateMainSettingsSuccessResponse"
+                UpdateMainSettingsFailedResponse:
+                  $ref: "#/components/examples/UpdateMainSettingsFailedResponse"
+        403:
+          description: The request was forbidden by ReadonlyREST's ACL
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ForbiddenRorResponse'
+              examples:
+                NoAclBlockWasMatchedForbiddenResponseExample:
+                  $ref: "#/components/examples/NoAclBlockWasMatchedForbiddenResponseExample"
+                ForbidBlockMatchedResponseExample:
+                  $ref: "#/components/examples/ForbidBlockMatchedResponseExample"
+                TestSettingsNotConfiguredForbiddenResponseExample:
+                  $ref: "#/components/examples/TestSettingsNotConfiguredForbiddenResponseExample"
+                ImpersonationNotSupportedResponseExample:
+                  $ref: "#/components/examples/ImpersonationNotSupportedResponseExample"
+                ImpersonationNotAllowedResponseExample:
+                  $ref: "#/components/examples/ImpersonationNotAllowedResponseExample"
+                RorIsNotStartedYetResponseExample:
+                  $ref: "#/components/examples/RorIsNotStartedYetResponseExample"
+                RorFailedToStartResponseExample:
+                  $ref: "#/components/examples/RorFailedToStartResponseExample"
+                CannotSaveClusterWideSettingsOnFreeLicense:
+                  $ref: "#/components/examples/CannotSaveClusterWideSettingsOnFreeLicense"
+
+components:
+  securitySchemes:
+    http:
+      type: http
+      scheme: basic
+
+  schemas:
+    ForbiddenRorResponse:
+      type: object
+      required:
+        - error
+        - status
+      properties:
+        error:
+          type: object
+          required:
+            - type
+            - reason
+            - due_to
+          properties:
+            type:
+              type: string
+              enum:
+                - forbidden_response
+            reason:
+              type: string
+              enum:
+                - forbidden
+            due_to:
+              type: array
+              description: |
+                It's an array of causes why ReadonlyREST's ACL forbade the request. 
+                WARNING: Please notice, that one element array ES used to print as a string!!!
+              items:
+                type: string
+                enum:
+                  - OPERATION_NOT_ALLOWED
+                  - FORBIDDEN_BY_BLOCK
+                  - TEST_SETTINGS_NOT_CONFIGURED
```

Comment:
IMO this one and the two related to impersonation won't occur in this case. Maybe we can skip them?

Reply:
yeah, these 3 will never occur in case of main settings